consensus/src/quorum_store/proof_manager.rs

```diff
@@ -269,14 +162,20 @@ impl ProofManager {
                                 .max_txns_after_filtering
                                 .saturating_sub(cur_unique_txns),
                         ));
-                        self.batch_queue.pull_batches(
-                            max_inline_txns_to_pull,
-                            excluded_batches
-                                .iter()
-                                .cloned()
-                                .chain(proof_block.iter().map(|proof| proof.info().clone()))
-                                .collect(),
-                        )
+                        let (inline_batches, inline_payload_size, _) =
+                            self.batch_proof_queue.pull_batches_with_transactions(
```

Comment:
So, we are basically pulling from batch_proof_queue thrice (`batch_proof_queue.pull_proofs`, `batch_proof_queue.pull_batches`, `batch_proof_queue.pull_batches_with_transactions`). 

What's the differrence between `batch_proof_queue.pull_batches` and `batch_proof_queue.pull_batches_with_transactions`. Both are for pulling inline batches without proofs right?

Reply:
If the block has space remaining after inserting proofs, then `batch_proof_queue.pull_batches` will then include as many batches without proofs as possible.

Reply:
Yes, we will still maintain inlining of transactions hence `pull_batches_with_transactions` under low TPS. `pull_batches` will come into play at higher TPS. A follow-up PR will add heuristics on what the composition of a block will be: how much proofs, how much inline batches and how much opt batches.

Reply:
Basically, my understanding is `batch_proof_queue.pull_batches_with_transactions` will not include any more batches. If there is enough space left in the block and if there is a batch without proof in ProofBatchQueue, then `batch_proof_queue.pull_batches` must have already included that batch.

Reply:
No, whether a batch should contain opt batches will be decided by `opt_batch_txns_pct` parameter. This will be 0 in case we don't want to include opt batches on the block. At that time, we can still include inline batches. There is a TODO in line 126 to clean this up.